### PR TITLE
Support Laravel 11

### PR DESCRIPTION
**Pull request by spawnia**
- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

Allow installation alongside Laravel 11.

**Breaking changes**

None.

### CODE REVIEW

**Comment by spawnia** [comment]
Requires https://github.com/BenSampo/laravel-enum/pull/353.

**Comment by elliottlawson** [comment]
@spawnia would like to pull this in on a Laravel 11 project, anything I can help with to move this forward?

**Comment by chojnicki** [comment]
> Requires [BenSampo/laravel-enum#353](https://github.com/BenSampo/laravel-enum/pull/353).

@spawnia Hmm but why lighthouse cannot work with PHP native enums? Even author suggests not to use that package.

File: composer.json [hunk 1]
```diff
@@ -50,31 +50,29 @@
         "fakerphp/faker": "^1.21",
         "google/protobuf": "^3.21",
         "larastan/larastan": "^2.6.1",
-        "laravel/framework": "^9 || ^10",
+        "laravel/framework": "^11",
```
Comment (spawnia) [inline composer.json:53]:
```suggestion
        "laravel/framework": "^9 || ^10 || ^11",
```

Just for testing to force the update.

**Comment by vanbosse** [comment]
@spawnia can we help you out in orde to push this forward?

**Comment by spawnia** [comment]
> @spawnia can we help you out in order to push this forward?

Sure, thank you for the offer. I just pushed a change to allow the CI pipelines to run in principle. You can send a merge request that targets this branch and fixes some of the issues in PHPUnit or PHPStan.

**Comment by nhedger** [comment]
I'm assuming you're dropping Lumen support. I can send a PR that removes references to lumen that PHPStan complains about if you're ok with it.

**Comment by spawnia** [comment]
> @spawnia I'm assuming you're dropping Lumen support. I can send a PR that removes references to Lumen that PHPStan complains about if you're ok with it.

I am not planning to drop support just yet. It is just impossible to install alongside Laravel 11 for testing, so I am dropping it as a dev dependency.

**Comment by chojnicki** [comment]
> I am not planning to drop support just yet.

I hope you meant "never" ❤️

**Comment by nhedger** [comment]
> > @spawnia I'm assuming you're dropping Lumen support. I can send a PR that removes references to Lumen that PHPStan complains about if you're ok with it.
> 
> I am not planning to drop support just yet. It is just impossible to install alongside Laravel 11 for testing, so I am dropping it as a dev dependency. We can just ignore the PHPStan errors.

Sounds good!

**Comment by spawnia** [comment]
Thank you @nhedger. Released with https://github.com/nuwave/lighthouse/releases/tag/v6.36.0.

**Comment by njoguamos** [comment]
Thank you all.

**Comment by nhedger** [comment]
Thank you @spawnia!

**Pull request merged**